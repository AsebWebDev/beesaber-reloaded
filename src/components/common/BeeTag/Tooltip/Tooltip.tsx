import Flag from 'react-country-flag';
import styled from 'styled-components';

import Avatar from '@/components/common/Avatar/Avatar';

import type { Bee } from '@/../sharedTypes';

const Container = styled.span`
  display: flex;
  flex-direction: column;
`;

type Props = {
  bee: Bee;
};

const Tooltip = ({ bee }: Props): JSX.Element => {
  const { avatar, playerName, rank, country } = bee;

  return (
    <Container>
      <Avatar avatar={avatar} playerName={playerName} />
      {playerName}
      <br />
      <Flag countryCode={country} />
      {rank}
      <br />
    </Container>
  );
};

export default Tooltip;
