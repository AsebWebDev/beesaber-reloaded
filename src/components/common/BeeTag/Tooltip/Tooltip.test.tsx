import { render } from '@testing-library/react';

import exampleBee from '@/testing/testData/exampleBee';

import Tooltip from './Tooltip';

describe('components/common/BeeTag/Tooltip', () => {
  it('should match snapshot', () => {
    const { container } = render(<Tooltip bee={exampleBee} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
