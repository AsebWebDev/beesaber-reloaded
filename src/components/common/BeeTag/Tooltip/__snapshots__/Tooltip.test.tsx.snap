// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/BeeTag/Tooltip should match snapshot 1`] = `
.c1 {
  margin-right: 1rem;
  max-width: 2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<span
  class="c0"
>
  <img
    alt="Avatar of player example Bee"
    class="c1"
    src="https://new.scoresaber.comsomeAvatarString"
  />
  example Bee
  <br />
  <span
    role="img"
    style="display: inline-block; font-size: 1em; line-height: 1em; vertical-align: middle;"
  >
    ðŸ‡ªðŸ‡³
  </span>
  4000
  <br />
</span>
`;
